Closes #3789 Add recommendation that developers using macOS CLI should install gpatch. (#4057)
az-digital/az_quickstart: CONTRIBUTING.md

that is downloaded when installing a site locally via Lando, or DDev.  See 
"Visual Studio Code Integration"

## Recommendation - macOS CLI Users

> Some patches do not apply via composer when running 'composer install' or 'composer update -W'. The solution is to install gpatch. 

Reviewer: ```suggestion
Some patches may not apply via composer when running `composer install` or `composer update` on macOS. This is because the version of the `patch` utility that is included with macOS differs from the version of GNU patch included with most Linux operating systems. A workaround for this problem is to install GNU patch using homebrew.
```

> It is recommended to premptively install this formulae and avoid this composer complication altogether.

Reviewer: ```suggestion
We recommended installing this formula preemptively on macOS to avoid this composer complication altogether.
```

> 'brew install gpatch' - [homebrew](https://brew.sh/)

Reviewer: ```suggestion
`brew install gpatch` - [homebrew](https://brew.sh/)
```

> `brew install gpatch` - [homebrew](https://brew.sh/)

Reviewer: ```suggestion
`brew install gpatch` - [homebrew formula](https://formulae.brew.sh/formula/gpatch)
```
